### PR TITLE
Support branches option for upcoming v16.0.0

### LINKED ISSUE #100: Support semantic-release v16.0.0-beta
The new branches functionality that landed in semantic-release [v16.0.0-beta.1](https://github.com/semantic-release/semantic-release/tree/v16.0.0-beta.1%40beta) is something I need but unfortunately this plugin doesn't yet support it.

I poked around a bit and noticed:

- The new `branch` attribute exists on the root of the context object.
- The branch attribute is [normalized to an object with {name: string} if `branches` is a string](https://github.com/semantic-release/semantic-release/blob/v16.0.0-beta.1%40beta/lib/branches/index.js#L16)

### PR DESCRIPTION
**Pull request by iamchrismiller**
Fix #100

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
     "get-stream": "^4.0.0",
     "git-log-parser": "^1.2.0",
     "nyc": "^13.1.0",
-    "semantic-release": "^15.0.0",
+    "semantic-release": "^16.0.0-beta.2",
```
Comment (pvdlg) [inline package.json]: No need to update the dependency here, please keep `^15.0.0`

File: package.json [hunk 2]
```diff
@@ -75,7 +75,7 @@
     "all": true
   },
   "peerDependencies": {
-    "semantic-release": ">=15.4.0 <16.0.0"
+    "semantic-release": "^16.0.0-beta.2"
```
Comment (pvdlg) [inline package.json]: This should be `>=16.0.0-beta <17.0.0`

Comment (iamchrismiller) [reply package.json]: 👍

File: package.json (hunk 1 above)
Comment (iamchrismiller) [reply package.json]: 👍

File: package.json [hunk 3]
```diff
@@ -37,7 +37,7 @@
     "get-stream": "^4.0.0",
     "git-log-parser": "^1.2.0",
     "nyc": "^13.1.0",
-    "semantic-release": "^15.0.0",
+    "semantic-release": "15.0.0",
```
Comment (pvdlg) [inline package.json]: It should be `^15.0.0` not `15.0.0`. You should not change this line at all

Comment (iamchrismiller) [reply package.json]: Sorry, coffee fully ingested now. I've added the caret back.

**Comment by pvdlg** [comment]
Release on `@beta` dist-tag as `7.1.0-beta.1`

**Pull request closed without merging**